burn-import/src/burn/node/base.rs

```diff
@@ -204,13 +207,14 @@ pub(crate) mod tests {
 
             #[derive(Module, Debug)]
             pub struct Model<B: Backend> {
-                _phantom: core::marker::PhantomData<B>,
+                phantom: core::marker::PhantomData<B>,
```

Comment:
I thought as well that it will come out as unused variable but since it's a field compiler does not mind. I used `phantom` for aesthetic reasons and I have seen this is how used. For example, the [examples in PhantomData](https://doc.rust-lang.org/beta/core/marker/struct.PhantomData.html) use `phantom` field name instead of `_phantom`.  So I didn't want to stand out =)

The reason that I am using `#[allow(unused_variables)]` for record in `with_init`. Unless I peak into the body's code, I can't decide if we should use `_record` or `record`. For simplicity and avoid error prone implementation, I opted to use `#[allow(unused_variables)]`. Of course doing so blankets the coverage, and in fact there is a variable unused in the body, we will not know. What do you think about this aspect? Should we check body's code if there is a reference to `record` argument? BTW, Dropout needs a field but does not need record, as you know.

Reply:
No let's keep `#[allow(unused_variable)]` for simplicity, we can revisit later if it becomes a problem.